### PR TITLE
[jsk_perception] add cmt_nodelet depending on libcmt

### PR DESCRIPTION
**Pull request by aginika**
DON'T MERGE YET MANY POINTS TO REPAIR.

libcmt_test

### CODE REVIEW

**Comment by garaemon** [comment]
Great!

**Comment by garaemon** [comment]
Please write a sample launch like:
`cmt tracker -> mask image -> mask pointcloud -> extract points -> euclidean segmentation -> extract nearest segment -> bounding box`

**Comment by garaemon** [comment]
and update README

**Comment by aginika** [comment]
add README and another subscriber to set the target regiont and reset tracking.

Still work on, DON'T MERGE THIS PR

**Comment by aginika** [comment]
publish mask image
publish_mask_image

**Comment by aginika** [comment]
@garaemon How can I do `extract nearest segment` ?

**Comment by garaemon** [comment]
good point, https://github.com/jsk-ros-pkg/jsk_demos/blob/master/jsk_2015_06_hrp_drc/drc_task_common/scripts/nearest_box_indices.py

**Comment by aginika** [comment]
@garaemon awesome, Thank you!

**Comment by aginika** [comment]
I have done around recognition.
But I have some problem around compile..

When I `catkin bt` in jsk_perception, unless I write as below, it will fail...
```
target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${OpenCV_LIBRARIES} libcmt.so)
## I adding libcmt.so directly
```
Although I add the libcmt dependency as catkin dependency and write in package.xml , If I DON'T add above line, ther error will appear like this... 
```
Built target white_balance_converter
[100%] Built target jsk_perception
Linking CXX executable /home/aginika/ros/hydro/devel/lib/jsk_perception/sliding_window_object_detector_trainer_node
/home/aginika/ros/hydro/devel/lib/libjsk_perception.so: undefined reference to `CMT::CMT()'
/home/aginika/ros/hydro/devel/lib/libjsk_perception.so: undefined reference to `CMT::processFrame(cv::Mat)'
/home/aginika/ros/hydro/devel/lib/libjsk_perception.so: undefined reference to `CMT::initialise(cv::Mat, cv::Point_<float>, cv::Point_<float>)'
collect2: ld returned 1 exit status
make[2]: *** [/home/aginika/ros/hydro/devel/lib/jsk_perception/sliding_window_object_detector_trainer_node] Error 1
make[1]: *** [CMakeFiles/sliding_window_object_detector_trainer_node.dir/all] Error 2
make: *** [all] Error 2

```


libctm_gif

**Comment by aginika** [comment]
Above Problem is also resolved. Thanks to @k-okada 
Now this package is also no problem , if libcmt is released.
SO please merge after libcmt merged. 
Thank you

File: jsk_perception/CMakeLists.txt [hunk 1]
```diff
@@ -92,6 +94,7 @@ macro(jsk_perception_nodelet _nodelet_cpp _nodelet_class _single_nodelet_exec_na
     jsk_perception_nodelet_sources jsk_perception_nodelet_executables)
 endmacro()
 jsk_perception_nodelet(src/edge_detector.cpp "jsk_perception/EdgeDetector" "edge_detector")
+jsk_perception_nodelet(src/cmt.cpp "jsk_perception/CMTNodelet" "cmt_node")
```
Comment (k-okada) [inline jsk_perception/CMakeLists.txt]: we should update jsk_percetpion_nodelet macro

**Comment by garaemon** [comment]
コンパイルできない気がする。
libcmtが3rdpartyにはいったので、
libcmtがみつかったらcmt_nodeletをコンパイルするように変更してください

**Comment by garaemon** [comment]
@aginika 

https://github.com/jsk-ros-pkg/jsk_recognition/pull/1186

途中までやったけどうまく行かなかったので任せます

**Comment by aginika** [comment]
了解です。

**Pull request closed without merging**